bundles/ch.elexis.global_inbox.core/src/ch/elexis/global_inbox/core/util/ImportOmnivoreInboxUtil.java

```diff
@@ -47,40 +52,54 @@ public ImportOmnivoreInboxUtil() {
 	 * @return the document id if import was successful, else <code>null</code>
 	 */
 	public @Nullable String tryImportForPatient(File file, String patientNo, String fileName) {
-		IQuery<IContact> query = CoreModelServiceHolder.get().getQuery(IContact.class);
-		query.and(ModelPackage.Literals.ICONTACT__CODE, IQuery.COMPARATOR.EQUALS, patientNo);
-		List<IContact> contacts = query.execute();
-		if (contacts.size() == 1) {
-			IContact contact = contacts.get(0);
-			IPatient pat = contact.asIPatient();
-			if (pat != null) {
-				if (!isFileOpened(file)) {
-					String cat = ImportOmnivoreInboxUtil.getCategory(file);
-					if (cat.equals("-") || cat.equals("??")) {
-						cat = null;
+		List<Patient> lPat = new Query(Patient.class, Patient.FLD_PATID, patientNo).execute();
+		if (lPat.size() == 1) {
+			if (!isFileOpened(file)) {
+				Patient pat = lPat.get(0);
+				String cat = ImportOmnivoreInboxUtil.getCategory(file);
+				if (cat.equals("-") || cat.equals("??")) { //$NON-NLS-1$ //$NON-NLS-2$
+					cat = null;
+				}
+				try {
+					long heapSize = Runtime.getRuntime().totalMemory();
+					long length = file.length();
+					if (length >= heapSize) {
+						logger.warn("Skipping " + file.getAbsolutePath() + " as bigger than heap size. (#3652)"); //$NON-NLS-1$ //$NON-NLS-2$
+						return null;
 					}
-					try {
-						long heapSize = Runtime.getRuntime().totalMemory();
-						long length = file.length();
-						if (length >= heapSize) {
-							logger.warn("Skipping " + file.getAbsolutePath() + " as bigger than heap size.");
-							return null;
-						}
-						IDocument newDocument = omnivoreDocumentStore.createDocument(pat.getId(), fileName, cat);
-						try (InputStream fileInputStream = Files.newInputStream(file.toPath())) {
-							omnivoreDocumentStore.saveDocument(newDocument, fileInputStream);
+					IDocument newDocument = omnivoreDocumentStore.createDocument(pat.getId(), fileName, cat);
+					String extension = getFileExtension(file);
+					if (extension != null
+							&& (newDocument.getMimeType() == null || newDocument.getMimeType().isEmpty())) {
+						MimeType mimetyp = MimeType.getByExtension(extension);
+						if (mimetyp != MimeType.undefined) {
+							newDocument.setMimeType(mimetyp.getContentType());
+						} else {
+							newDocument.setMimeType(file.getName());
 						}
-						file.delete();
-						return newDocument.getId();
-					} catch (Exception ex) {
-						logger.warn("Fehler " + ex.getMessage());
 					}
+					try (InputStream contentStream = new FileInputStream(file)) {
+						omnivoreDocumentStore.saveDocument(newDocument, contentStream);
+					}
+					file.delete();
+					return newDocument.getStoreId();
```

Comment:
Das hier ist nicht die id des Dokuments sondern die des store, also immer omnivore. Bitte anpassen das die document id zurückgegeben wird, wie in der java doc spezifiziert.